core/src/main/java/org/apache/iceberg/util/JsonUtil.java

```diff
@@ -134,6 +147,18 @@ public static String getStringOrNull(String property, JsonNode node) {
         .build();
   }
 
+  public static void writeIntegerIfExists(String key, Integer value, JsonGenerator generator) throws IOException {
```

Comment:
In other places with a similar pattern, we move the whole boolean to the caller so that this is a bit more flexible:

```java
  JsonUtil.writeIntegerIf(v != null, "min-snapshots-to-keep", v, generator);
```